src/startupscriptgenerator/src/dotnetcore/scriptgenerator.go

```diff
@@ -42,6 +42,16 @@ func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuil
 	}
 
 	dotnetBinary := "/usr/share/dotnet/dotnet"
+
+	preRunScriptPath := common.GetPathEnvironmentVariable(consts.AppServicePreRunScriptFilePathEnvVarName)
+	if preRunScriptPath != "" && !common.PathExists(dotnetBinary) {
+		scriptBuilder.WriteString("echo 'Running the user provided pre-run command within pre-run script... ")
+		preRunScriptFileDir := filepath.Dir(preRunScriptPath)
+		scriptBuilder.WriteString("cd \"" + preRunScriptFileDir + "\"\n")
+		preRunCommand := "bash \"" + preRunScriptPath + "\""
+		scriptBuilder.WriteString(preRunCommand + "\n")
```

Comment:
I think this new block should be added after line # 58 (i.e after we install the platform sdk). For example, a user might try to use some platform tools to do something in their pre-run commands.